master/buildbot/reporters/pushover.py

```diff
@@ -0,0 +1,294 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+from future.utils import iteritems
+from future.utils import string_types
+
+from twisted.internet import defer
+from twisted.python import log as twlog
+
+from buildbot import config
+from buildbot import interfaces
+from buildbot import util
+from buildbot.process.properties import Properties
+from buildbot.process.results import CANCELLED, EXCEPTION, FAILURE, SUCCESS, WARNINGS
+from buildbot.process.results import Results
+from buildbot.reporters import utils
+from buildbot.reporters.message import MessageFormatter as DefaultMessageFormatter
+from buildbot.reporters.message import MessageFormatterMissingWorker
+from buildbot.util import service
+
+try:
+    from pushover import Client
```

Comment:
pushover has a rest api. Please use this rest api rather than a python wrapper. this is best integrated with twisted.